guide/src/python_from_rust.md

```diff
@@ -230,3 +230,48 @@ def leaky_relu(x, slope=0.01):
 
 [`Python::run`]: https://docs.rs/pyo3/latest/pyo3/struct.Python.html#method.run
 [`py_run!`]: https://docs.rs/pyo3/latest/pyo3/macro.py_run.html
+
+## Need to use a context manager from Rust?
+
+Use context managers by directly invoking `__enter__` and `__exit__`.
+
+```rust
+use pyo3::prelude::*;
+use pyo3::types::PyModule;
+
+fn main() {   
+    Python::with_gil(|py| {
+        let custom_manager = PyModule::from_code(py, r#"
+class House(object):
+    def __init__(self, address):
+        self.address = address
+    def __enter__(self):
+        print(f"Welcome to {self.address}!")
+    def __exit__(self, type, value, traceback):
+        if type:
+            print(f"Sorry you had {type} trouble at {self.address}")
+        else:
+            print(f"Thank you for visiting {self.address}, come again soon!")
+
+        "#, "objects.py", "objects").unwrap();
+        
+        let house = custom_manager.call1("House", ("123 Main Street",)).unwrap();
```

Comment:
Should be `call_function1`, because we actually just deprecated `call1` on master.

@birkenfeld strictly speaking this isn't a function, so maybe `call_attribute` is better? Although I wonder if actually we should plan to remove `module::call{_function}` etc. completely and instead just ask users to do `.getattr("...")?.call1("...")?` etc? It's not much longer and doesn't suffer with naming ambiguity.

Reply:
I think you must mean `call_method1`, so I've changed it to that!

Reply:
See https://github.com/PyO3/pyo3/commit/e065f9b5170f76a529aaf5bad25256ed5b7d75f7 - `call_function1` is very hot off the press :)

Reply:
> 
> 
> See [e065f9b](https://github.com/PyO3/pyo3/commit/e065f9b5170f76a529aaf5bad25256ed5b7d75f7) - `call_function1` is very hot off the press :)

My understanding is that the `call_function1` change will require a larger-scale rewrite of the documentation, which is out of scope for this specific pull request.  For example: call is still used earlier in this same document.

I would propose leaving this as-is, and doing the broader change to `call_function1` in a separate pull request that can cover a wide scope and capture all the appropriate uses of `call` and otherwise that should be changed.

Reply:
Oops, you updated the documentation in that commit. Ignore earlier message if you saw it! I'll make the change!